### ISSUE TITLE
object_usage_linter fails to account for withr-defined names

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2714]
```r
lint(linters = object_usage_linter(), R"{
foo <- function(f) {
  withr::with_connection(list(myCon = f), writeLines(letters, myCon))
}
}")
# <text>:3:63: warning: [object_usage_linter] no visible binding for global variable 'myCon'
#   withr::with_connection(list(myCon = f), writeLines(letters, myCon))
#                                                               ^~~~~
```

Obviously {codetools} does not (and probably will not) support `myCon` being defined as valid in this context by {withr}, but we can supplement to remove this false positive as we do in a few similar cases. {withr} is popular enough (100+ revdeps on CRAN) that this special case is warranted.
